Docs update

Yoshanuikabundi opened the pull request:
## Description
Add working documentation!

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] Automated API docs
  - [x] OpenFF Sphinx theme
  - [x] Up-to-date installation guide
  - [x] Docstrings for public modules
  - [x] Getting started guide
  - [x] Additional book-style docs'
      - [x] Schemas/pydantic
      - [x] Configuring the workflow factory
      - [x] BespokeExecutor
      - [x] Reading results
      - [x] The BespokeFit Command Line Interface
  - [ ] Update readme
  - [x] Doctests
      - [x] Setup
      - [x] Test (requires rebase after API change)
  - [x] Update with changes from upcoming API change
      - [x] Rebase
      - [x] Update user's guide
  - [ ] Return RTD latest version to main branch before merging

## Status
- [ ] Ready to go

j-wags commented:
I'll give this a preliminary once-over (even though I know it's waiting on the API break). I've been meaning to read into bespokefit so this would be a great chance to get two birds with one stone!

j-wags commented on docs/userguide/04_bespoke-executor.md:
```diff
@@ -0,0 +1,141 @@
+This might be a local workstation or a remote HPC cluster. It then provides a
+server to which jobs can be submitted. Submitted jobs are placed in a queue
+until resources become available, at which point the executor runs them.
+BespokeExecutor can break a workflow down into stages and run parts in
+parallel with parts from other workflows so that even serial steps can take
+advantage of the host platform's parallelism. This means that several workflows
```
This sentence is kinda hard to read, though I can't quite suggest a better wording for it.

Yoshanuikabundi replied:
Rephrased. Thanks!

j-wags commented on docs/userguide/04_bespoke-executor.md:
```diff
@@ -0,0 +1,141 @@
+//  from openff.bespokefit.executor import BespokeExecutor
+//  from openff.bespokefit.workflows import BespokeWorkflowFactory
+//  from openff.toolkit.topology import Molecule
+//  target_molecule = Molecule.from_smiles("C(C(=O)O)N") # Glycine
+//  workflow = factory.optimization_schema_from_molecule(
+//      molecule=target_molecule,
+//  )
```
These lines may not actually be getting used

Yoshanuikabundi replied:
Good point! Removed

j-wags commented on docs/userguide/04_bespoke-executor.md:
````diff
@@ -0,0 +1,141 @@
+```
+
+The `--input` switch accepts most commonly used chemistry formats. OpenFF
+recommends SDF as it includes all the information needed to unambiguously
````
```suggestion
recommends SDF or MOL as they include all the information needed to unambiguously
```

j-wags reviewed (commented):
Just did a quick read-through. This looks really great - As a newbie I think I really got a good understanding of what's happening. I commented a bunch but it's almost entirely nitpicking. In this pass, I held off from actually running commands since the API change is coming. 

So I think this passes the "positive space" test quite well -- the stuff it DOES describe is well-written and understandable. When you're ready for more feedback, Simon and Josh H can give feedback on the "negative space" check - basically if there's some prose here that's mistaken or an area that missing. But I assume you'll do that after you update this for the API break and have marked it ready for review.

Yoshanuikabundi commented on docs/userguide/02_schemas.md:
```diff
@@ -0,0 +1,81 @@
+(schemas_chapter)=
+# Schemas in BespokeFit
+
+BespokeFit uses [Pydantic] to "Define how data should be in pure, canonical
+python; validate it with *pydantic*." Pydantic helps BespokeFit provide clear
```
Yeah you're right; I've removed the quote and put the idea in my own words.

Yoshanuikabundi commented on docs/userguide/02_schemas.md:
````diff
@@ -0,0 +1,81 @@
+factory = BespokeWorkflowFactory()
+factory.schema()
+```
+
````
I don't, because it'll fall out of date, and also because schemas are very long, and there's a link to the API documentation that has the info in the previous paragraph. I'm also yet to implement the ability to include output in my doctest setup.

Yoshanuikabundi commented on docs/userguide/03_workflow-factory.md:
```diff
@@ -0,0 +1,182 @@
+
+### QC Generation
+
+BespokeFit uses target schemas to specify how to fit parameters. These schemas are
+responsible for generating the specific tasks used to generate reference data,
+and also describe how significant each target is in terms of the accuracy of the
```
Thanks for the catch on repeating "generate". The phrasing here is supposed to say that the schema's responsibility is the generate tasks, but they also do this other important thing as part of that responsibility. I've rephrased it into two sentences to make it easier to parse.

j-wags replied:
Great. Thanks!

Yoshanuikabundi commented on docs/userguide/03_workflow-factory.md:
````diff
@@ -0,0 +1,182 @@
+optimizer = ForceBalanceSchema()
+```
+
+The optimizer also needs an initial, unconstrained force field to use as a starting
+point. This should be the filename of a force field in [offxml format]:
````
That's a good idea; we could also discuss how to add constraints back in to an unconstrained force field. Can we discuss this at our next check-in? I'm not sure I'm across all the details.

j-wags replied:
Yes, I'll add it to our agenda!

davidlmobley commented:
Following

Yoshanuikabundi commented:
Hey @jthorton, I've rebased on main and updated the user guide for recent changes. Would be great if you could read through things and let me know if anything is

1. Missing
2. Incorrect
3. Unnecessarily difficult

It'd also be good to have a call at some point to talk about how stages work and possibly going over the API and thinking about where we can use privacy and exports to simplify it before we release.

Also - is there some sort of mock QC engine for tests I could use in the doctests? At the moment I test all my code examples with XTB for speed but there's not really any reason not to use a mock engine.

Current documentation is rendered [here](https://openff-bespokefit.readthedocs.io/en/latest/index.html)

j-wags commented:
Merging this is pending review from a code author (or a go-ahead to merge without author approval!)

SimonBoothroyd commented:
@Yoshanuikabundi thanks so much for putting this together! This is a great starting point for sure. I'm going to go ahead and get this merged in so that it's easier to keep the docs in sync with the code changes that are going to be made this week such as #116 .
